### PR TITLE
feat: trials comparison table in experiment comparison [WEB-993]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Screenshot 2023-06-08 at 14-49-18 Project Details - Determined

This involved a fair bit of refactoring and cleaning up old code, but if you spot anything I missed that would be an easy fix let me know.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/TrialsComparisonModal.tsx [hunk 1]
```diff
@@ -111,179 +129,172 @@ const TrialsComparisonTable: React.FC<TableProps> = ({
...
-  );
+  const prevHps = usePrevious(hyperparameterNames, []);
 
   useEffect(() => {
-    setSelectedHyperparameters(hyperparameterNames);
-  }, [hyperparameterNames]);
+    setSelectedHyperparameters((prevSelectedHps) =>
+      isEqual(prevSelectedHps, prevHps) ? hyperparameterNames : prevSelectedHps,
+    );
+  }, [hyperparameterNames, prevHps]);
```
Comment (gt2345) [inline webui/react/src/pages/ExperimentDetails/TrialsComparisonModal.tsx:197]: Why do we want to track previous value of hyper parameters and metrics?

**Review by gt2345 — APPROVED** [review]
Works as expected. LGTM.

**Comment by alexjdetermined** [comment]
* seeing an error pop up often re "Experiment metric name stream failed". 
image
* idling with details tab open im seeing the amount of "metrics" change with some appearing and disappearing at random intervals. 

Revisions
1. Change tab label from "Configurations" to "Details"
2. Change "Experiment 104" to the Experiment name
3. Is it possible to add two rows under "state" for Experiment ID and Trial ID?

**Comment by alexjdetermined** [comment]
@ClaireNeveu can you weigh in on this? This is prob out of scope for this ticket but desired, should we add this as another ticket?

Desired updates
1. Include corresponding colored checkbox next to the experiment name in the column header in details table.
2. clicking that will unselect from table

**Comment by ClaireNeveu** [comment]
@alexjdetermined Oof, I can see how that might require a big change to the code since you're turning a unidirectional mapping into a bidirectional mapping. From a technical perspective (without looking at the codebase) that sounds like something we wouldn't want to do.

From a product/design standpoint it also seems like more of a nice-to-have that something necessary so I would advise keeping that out of this PR and probably out of V1 entirely (sounds like a good source of bugs so I'd rather make that change from a known good state).

**Review by alexjdetermined — APPROVED** [review]
LGTM

**Pull request merged**